module/zfs/zil.c

```diff
@@ -1838,17 +1834,47 @@ zil_lwb_write_issue(zilog_t *zilog, lwb_t *lwb)
 		nlwb = zil_alloc_lwb(zilog, bp, slog, txg, TRUE);
 	}
 
+	lwb->lwb_state = LWB_STATE_ISSUED;
```

Comment:
I'm still reading the changes, so this might be a silly question..

But does it make sense to set the state to `ISSUED` in this function, still? E.g. since we don't actually issue the write until the `zil_lwb_write_issue` function?

Maybe it'll make more sense as I read how this function, and `zil_lwb_write_issue`, are called.. still reading..

Reply:
We definitely must set lwb_issued_txg and bump zl_lwb_inflight before committing transaction below, or we may end up syncing with zio still in flight.  lwb_state may be less critical, but my original idea was that nobody should care that zio is not really issued yet.  Later there appeared some edge cases, but I left it as-is.